Stellar-ledger.x

```diff
@@ -430,6 +442,11 @@ struct TransactionMetaV3
 
     Hash hashes[3];                     // stores sha256(txChangesBefore, operations, txChangesAfter),
                                         // sha256(events), and sha256(txResult)
+
+    // Diagnostics events that are not hashed. One list per operation.
+    // This will contain all contract and diagnostic events. Even ones
+    // that were emitted in a failed contract call.
+    OperationDiagnosticEvents diagnosticEvents<>;
```

Comment:
Does that mean this is a subset of `OperationEvents`, or they're disjoint?

Reply:
o shoot okay so this at least doubles the size of the event-related meta, right? why can't `TransactionMetaV3.events` mark diagnostic events w/ their type accordingly? is it because you'd need to make `ContractEvent` a union to include the extraneous `inSuccessfulContractCall` field?

Reply:
`events` will be hashed into the ledger and will be part of the protocol. We want diagnostic events to not affect the protocol, which is why we keep it separate.